spec/api-web-contents-spec.js

```diff
@@ -799,4 +800,50 @@ describe('webContents module', () => {
       w.loadURL('about:blank')
     })
   })
+
+  describe('takeHeapSnapshot()', () => {
+    const filePath = path.join(remote.app.getPath('userData'), 'test.heapsnapshot')
+
+    afterEach(done => {
+      fs.unlink(filePath, () => done())
+    })
+
+    it('works with sandboxed renderers', async () => {
+      w.destroy()
+      w = new BrowserWindow({
+        show: false,
+        webPreferences: {
+          sandbox: true
+        }
+      })
+
+      w.loadURL('about:blank')
+      await emittedOnce(w.webContents, 'did-finish-load')
+
+      await w.webContents.takeHeapSnapshot(filePath)
+    })
+
+    it('fails with invalid file path', async () => {
+      w.destroy()
+      w = new BrowserWindow({
+        show: false,
+        webPreferences: {
+          sandbox: true
+        }
+      })
+
+      w.loadURL('about:blank')
+      await emittedOnce(w.webContents, 'did-finish-load')
+
+      let message
+
+      try {
+        await w.webContents.takeHeapSnapshot('')
```

Comment:
We can check this promise directly, without `await`:

```js
const heapSnapshotTaken = w.webContents.takeHeapSnapshot('')
return expect(heapSnapshotTaken)
    .to.be.eventually.rejectedWith(Error, 'takeHeapSnapshot failed')
```

Reply:
done